### ISSUE TITLE
Temp files should be created in a temp directory

### ISSUE THREAD

**Issue by axelfontaine** [testcontainers/testcontainers-java#423]
Currently temp files are created in the current directory. This leads to a mess in case the JVM crashes. Instead the files should be created in a subdirectory of the temp directory.

**Comment by bsideup** [comment]
Hi @axelfontaine,

Unfortunately, Docker doesn't allow mounts outside of user home directory, at least on Mac / Windows, this is why we do that. Any idea how to improve?

**Comment by axelfontaine** [comment]
The file I had was actually created in my source directory on the host Windows x64 machine and it doesn't lie under my user home. I have a feeling this temp file may not have been created by Docker, but by testcontainers itself. (I deleted before filing this issue and forgot to write down the exact name, but it was something like `.testcontainers.tmp.something`

**Comment by axelfontaine** [comment]
The culprit is the first line in `org.testcontainers.utility.MountableFile.extractClassPathResourceToTempLocation()` :

```java
File tmpLocation = new File(".testcontainers-tmp-" + Base58.randomString(5));
```

**Comment by arcuri82** [comment]
Having this same issue, where my folders get polluted by these tmp files. If for any reason the TMP folder of the OS cannot be used, a better default location than "." could be "./target" (at least for Maven builds)

**Comment by kiview** [comment]
Has been fixed with #450.
